src/dmd/backend/cg87.d

```diff
@@ -6,111 +6,134 @@
-#define PI              3.14159265358979323846
-#define LOG2            0.30102999566398119521
-#define LN2             0.6931471805599453094172321
-#define LOG2T           3.32192809488736234787
-#define LOG2E           1.4426950408889634074   /* 1/LN2                */
+enum PI            = 3.14159265358979323846;
+enum LOG2          = 0.30102999566398119521;
+enum LN2           = 0.6931471805599453094172321;
+enum LOG2T         = 3.32192809488736234787;
+enum LOG2E         = 1.4426950408889634074;   // 1/LN2
 
-#define FWAIT   0x9B            /* FWAIT opcode                         */
+enum FWAIT = 0x9B;            // FWAIT opcode
 
 /* Mark variable referenced by e as not a register candidate            */
-#define notreg(e)       ((e)->EV.sp.Vsym->Sflags &= ~GTregcand)
+uint notreg(elem* e) { return e.EV.Vsym.Sflags &= ~GTregcand; }
 
 /* Generate the appropriate ESC instruction     */
-#define ESC(MF,b)       (0xD8 + ((MF) << 1) + (b))
-enum MF
+ubyte ESC(uint MF, uint b) { return cast(ubyte)(0xD8 + (MF << 1) + b); }
+enum
 {       // Values for MF
         MFfloat         = 0,
         MFlong          = 1,
         MFdouble        = 2,
         MFword          = 3
-};
+}
 
-NDP _8087elems[8];              // 8087 stack
+__gshared
+{
+NDP[8] _8087elems;              // 8087 stack
 NDP ndp_zero;
 
 int stackused = 0;              /* number of items on the 8087 stack    */
+}
 
 /*********************************
  */
 
 struct Dconst
 {
     int round;
-    symbol *roundto0;
-    symbol *roundtonearest;
-};
-
-static Dconst oldd;
+    Symbol *roundto0;
+    Symbol *roundtonearest;
+}
 
-#define NDPP    0       // print out debugging info
-#define NOSAHF  (I64 || config.fpxmmregs)     // can't use SAHF instruction
+private __gshared Dconst oldd;
 
-static void loadComplex(CodeBuilder& cdb,elem *e);
-static void opmod_complex87(CodeBuilder& cdb,elem *e,regm_t *pretregs);
-static void opass_complex87(CodeBuilder& cdb,elem *e,regm_t *pretregs);
+enum NDPP = 0;       // print out debugging info
+bool NOSAHF() { return I64 || config.fpxmmregs; }     // can't use SAHF instruction
 
-#define CW_roundto0             0xFBF
-#define CW_roundtonearest       0x3BF
+enum CW_roundto0 = 0xFBF;
+enum CW_roundtonearest = 0x3BF;
 
-static void genrnd(CodeBuilder& cdb, short cw);
 
 /**********************************
  * When we need to temporarilly save 8087 registers, we record information
  * about the save into an array of NDP structs:
  */
 
-NDP *NDP::save = NULL;
-int NDP::savemax = 0;           /* # of entries in NDP::save[]          */
-int NDP::savetop = 0;           /* # of entries used in NDP::save[]     */
+/+
+struct NDP
+{
+    elem *e;                    // which elem is stored here (NULL if none)
+    uint offset;                // offset from e (used for complex numbers)
+
+    __gshared NDP *save;
+    __gshared int savemax;         // # of entries in save[]
+    __gshared int savetop;         // # of entries used in save[]
+}
++/
```

Comment:
Could be removed as this is defined in `code_x86` now.

Reply:
I wasn't sure what was best to do with that.